benchmarks/src/locale_classic.cpp

```diff
@@ -8,7 +8,8 @@ using namespace std;
 // GH-3048 <locale>: Double-checked locking for locale::classic
 void BM_locale_classic(benchmark::State& state) {
     for (auto _ : state) {
-        benchmark::DoNotOptimize(locale::classic());
+        auto v = locale::classic();
+        benchmark::DoNotOptimize(v);

```

Comment:
temporary was **const** rvalue

Reply:
Attempting to add the warnings in google/benchmark#1993.

Reply:
Would be very good. It would mean that we don't need to find a way to add clang-cl compilation to CI, as MSVC would spot most of the issues fixed here.